File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    vector<int> v = {0, 1};
    int answer = 0;
    if(n == 0 || n == 1) return n;
    for(int i = 2; i <= n; i++){
        v.push_back((v[i-1]+v[i-2])%1234567);
    }
    return v.back()%1234567;
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    n1 = 0
    n2 = 1
    if n == 0 : return 0
    if n == 1 : return 1
    for i in range(2, n+1):
        n1, n2 = n2, n1 + n2
    return n2 % 1234567